### PR TITLE
If memory utilisation is not found to be stable, return the median sa…

### PR DESCRIPTION
**Pull request by baldersheim**
…mple.
@geirst PR

### CODE REVIEW

File: staging_vespalib/src/vespa/vespalib/util/process_memory_stats.h [hunk 1]
```diff
@@ -37,6 +36,7 @@ class ProcessMemoryStats
     uint64_t getMappingsCount() const { return _mappings_count; }
     bool similarTo(const ProcessMemoryStats &rhs, uint64_t sizeEpsilon) const;
     vespalib::string toString() const;
+    bool operator < (const ProcessMemoryStats & rhs) const { return _mapped_rss < rhs._mapped_rss; }
```
Comment (geirst) [inline staging_vespalib/src/vespa/vespalib/util/process_memory_stats.h]: We are using _anonymous_rss when calculating memory used ratio, so I guess we should use that when ordering as well:
`
double
DiskMemUsageFilter::getMemoryUsedRatio(const Guard &guard) const
{
    (void) guard;
    uint64_t unscaledMemoryUsed = _memoryStats.getAnonymousRss();
    return static_cast<double>(unscaledMemoryUsed) / _hwInfo.memory().sizeBytes();
}
`

**Review by geirst — CHANGES_REQUESTED** [review]
Please have a look at my comment

**Comment by geirst** [comment]
@toregge please review as well

File: staging_vespalib/src/vespa/vespalib/util/process_memory_stats.h (hunk 1 above)
Comment (baldersheim) [reply staging_vespalib/src/vespa/vespalib/util/process_memory_stats.h]: Ok, I assumed that mapped was both file backed and anonymous. @toregge what say you.

**Comment by baldersheim** [comment]
@geirst Now it should be fixed to your liking.

**Review by toregge — APPROVED** [review]
LGTM

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**